cruise-control/src/main/java/com/linkedin/kafka/cruisecontrol/executor/ExecutionTaskManager.java

```diff
@@ -291,19 +185,17 @@ public synchronized void addExecutionProposals(Collection<ExecutionProposal> pro
                                                  ReplicaMovementStrategy replicaMovementStrategy) {
     _executionTaskPlanner.addExecutionProposals(proposals, cluster, replicaMovementStrategy);
     for (ExecutionProposal p : proposals) {
-      _inProgressReplicaMovementsByBrokerId.putIfAbsent(p.oldLeader(), 0);
-      for (int broker : p.replicasToAdd()) {
-        _inProgressReplicaMovementsByBrokerId.putIfAbsent(broker, 0);
-      }
+      p.replicasToMoveByBroker().keySet().forEach(broker -> _inProgressIntraBrokerReplicaMovementsByBrokerId.putIfAbsent(broker, 0));
+      _inProgressInterBrokerReplicaMovementsByBrokerId.putIfAbsent(p.oldLeader().brokerId(), 0);
+      p.replicasToAdd().forEach(r -> _inProgressInterBrokerReplicaMovementsByBrokerId.putIfAbsent(r.brokerId(), 0));
     }
     // Set the execution mode for tasks.
     _executionTaskTracker.setExecutionMode(_isKafkaAssignerMode);
 
     // Add pending proposals to indicate the phase before they become an executable task.
-    _executionTaskTracker.taskForReplicaAction(ExecutionTask.State.PENDING)
-                         .addAll(_executionTaskPlanner.remainingReplicaMovements());
-    _executionTaskTracker.taskForLeaderAction(ExecutionTask.State.PENDING)
-                         .addAll(_executionTaskPlanner.remainingLeadershipMovements());
+    _executionTaskTracker.initializeTask(_executionTaskPlanner.remainingInterBrokerReplicaMovements(), TaskType.INTER_BROKER_REPLICA_ACTION);
+    _executionTaskTracker.initializeTask(_executionTaskPlanner.remainingIntraBrokerReplicaMovements(), TaskType.INTRA_BROKER_REPLICA_ACTION);
+    _executionTaskTracker.initializeTask(_executionTaskPlanner.remainingLeadershipMovements(), TaskType.LEADER_ACTION);
```

Comment:
Does `initializeTask` handle `LEADER_ACTION`? It is not clear from the content of that function -- why do we need this part?

Reply:
good catch, `initializeTask` is to initialize the `dataToMoveInMB` field of `ExecutionTaskTracker`, so `LEADER_ELECTION` is unrelated here.